src/agent/adu_core_interface/tests/adu_core_interface_ut.cpp

```diff
@@ -295,8 +290,7 @@ TEST_CASE_METHOD(TestCaseFixture, "AzureDeviceUpdateCoreInterface_ReportContentI
             << R"(})"
          << R"(})";
     // clang-format on
-    // NOLINTNEXTLINE(cppcoreguidelines-pro-type-reinterpret-cast)
-    CHECK_THAT(reinterpret_cast<const char*>(g_SendReportedStateValues.reportedState), Equals(strm.str()));
+    CHECK_THAT(g_SendReportedStateValues.reportedState, Equals(strm.str()));
```

Comment:
>CHECK_THAT [](http://example.com/codeflow?start=4&length=10)

nit: if these are both strings now, CHECK( ... === ...) would be clearer throughout